src/schedule/zephyr_twb_schedule.c

```diff
@@ -0,0 +1,472 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2024 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@intel.com>
+ */
+
+#include <rtos/task.h>
+#include <stdint.h>
+#include <sof/lib/uuid.h>
+#include <sof/schedule/schedule.h>
+#include <sof/schedule/ll_schedule.h>
+#include <sof/schedule/twb_schedule.h>
+#include <sof/schedule/ll_schedule_domain.h>
+#include <sof/trace/trace.h>
+#include <rtos/wait.h>
+#include <rtos/interrupt.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/kernel/thread.h>
+
+LOG_MODULE_REGISTER(twb_schedule, CONFIG_SOF_LOG_LEVEL);
+
+/* e93326d8-0d14-4bf0-bcb9-e063d3d80136 */
+DECLARE_SOF_UUID("twb-schedule", twb_sched_uuid, 0xe93326d8, 0x0d14, 0x4bf0,
+		 0xbc, 0xb9, 0xe0, 0x63, 0xd3, 0xd8, 0x01, 0x36);
+
+DECLARE_TR_CTX(twb_tr, SOF_UUID(twb_sched_uuid), LOG_LEVEL_INFO);
+
+struct scheduler_twb_data {
+	struct list_item tasks;		/* list of active TWB tasks */
+	struct task ll_tick_src;	/* LL task - source of TWB tick */
+};
+
+struct task_twb_data {
+	k_tid_t thread_id;		/* zephyr thread ID */
+	k_thread_stack_t __sparse_cache *p_stack; /* pointer to thread stack */
+	int32_t thread_prio;		/* thread default priority */
```

Comment:
size_t

Reply:
Thread priority could be a negative value and signed int is used by kernel functions.
On the other hand preemptible thread (which is required for TWB) has a non-negative priority value so I just added priority value check in task init func.